fundingmanager.go

```diff
@@ -754,3 +785,46 @@ func (f *fundingManager) handleInitFundingMsg(msg *initFundingMsg) {
 	)
 	msg.peer.queueMsg(fundingReq, nil)
 }
+
+// processErrorGeneric sends a message to the fundingManager allowing it
+// to process the occurred generic error.
+func (f *fundingManager) processErrorGeneric(err *lnwire.ErrorGeneric,
+	peer *peer) {
+	f.fundingMsgs <- &fundingErrorMsg{err, peer}
+}
+
+// handleErrorGenericMsg process the error which was received from remote peer,
+// depends on the type of error we should do different clean up steps and
+// inform user about it.
+func (f *fundingManager) handleErrorGenericMsg(fmsg *fundingErrorMsg) {
+	if fmsg.err.ErrorID == lnwire.ErrorMaxPendingChannels {
+		peerID := fmsg.peer.id
+		chanID := fmsg.err.PendingChannelID
+
+		f.resMtx.RLock()
+		resCtx, ok := f.activeReservations[peerID][chanID]
+		f.resMtx.RUnlock()
+
+		if !ok {
+			resCtx.err <- fmt.Errorf("ErrorGeneric error " +
+				"was returned from remote peer for channel "+
+				"(id: %v), but it can't be found and thereby "+
+				"can't be canceled.", chanID)
+		}
+
+		if err := resCtx.reservation.Cancel(); err != nil {
+			resCtx.err <- fmt.Errorf("Remote peer responded "+
+				"with: Number of pending channels exceed "+
+				"maximum, but we can't cancel the reservation "+
+				"- %v", err)
+		} else {
+			resCtx.err <- grpc.Errorf(OpenChannelFundingError,
+				"Remote peer responded with: Number of "+
+					"pending channels exceed maximum")
+		}
+
+		f.resMtx.RLock()
```

Comment:
Since you're performing a delete from the map here (a write), you need to grab the full lock (`.Lock()`) instead of only the read lock.